testutil/deepequal.go

```diff
@@ -0,0 +1,129 @@
+package testutil
+
+import "reflect"
+
+// DeepEqual is similar to reflect.DeepEqual, but treats
+// nil and []T{} as equal.
```

Comment:
This is true for maps as well right? Maybe something like "…but treats empty maps and slices as equal to nil" to use conventional nomenclature.